### ISSUE TITLE
Change systemd execstart to yml conf

### ISSUE THREAD

**Issue by P4R4DiSi4C** [fluent/fluent-bit#9714]
Currently, '.conf' is hardcoded in the following file: https://github.com/fluent/fluent-bit/blob/master/init/systemd.in
Is there any plan to change it to '.yml' so we avoid the need of overriding the systemd configuration ?

```
[Unit]
Description=@FLB_PROG_NAME@
Documentation=https://docs.fluentbit.io/manual/
Requires=network.target
After=network.target

[Service]
Type=simple
EnvironmentFile=-/etc/sysconfig/@FLB_OUT_NAME@
EnvironmentFile=-/etc/default/@FLB_OUT_NAME@
ExecStart=@CMAKE_INSTALL_FULL_BINDIR@/@FLB_OUT_NAME@ -c /@CMAKE_INSTALL_SYSCONFDIR@/@FLB_OUT_NAME@/@FLB_OUT_NAME@.conf
Restart=always

[Install]
WantedBy=multi-user.target
```

**Comment by patrick-stephens** [comment]
Yes although we will have to be careful of impact on existing users who may be relying on overriding the conf file. It would have to be documented explicitly and maybe as a breaking change e.g. in 4.0 release @edsiper ?

YAML format was only completed for 3.2 as well so previous versions are still supported.

**Comment by chadmyers** [comment]
@P4R4DiSi4C How do you override the system configuration to use yaml?

**Comment by P4R4DiSi4C** [comment]
> @P4R4DiSi4C How do you override the system configuration to use yaml?

I have the following Ansible task:
```- name: "Override systemd configuration if needed."
  ansible.builtin.template:
    src: fluent-bit-override.conf.j2
    dest: "/etc/systemd/system/fluent-bit.service.d/override.conf"
    mode: '0644'
  when: fluentbit_systemd_override is defined
```

With the following template:

```
{{ ansible_managed | comment }}

[Service]
{% for key, value in fluentbit_systemd_override.items() %}
{% if key == 'ExecStart' %}
{{ key }}=
{% endif %}
{{ key }}={{ value }}
{% endfor %}
```
And set the following value:
```
fluentbit_systemd_override:
  ExecStart: '/opt/fluent-bit/bin/fluent-bit -c //etc/fluent-bit/fluent-bit.yml'
```

Which results in a `override.conf` file being created in the folder `/etc/systemd/system/fluent-bit.service.d/` with the following content:
```
#
# Ansible managed
#

[Service]
ExecStart=
ExecStart=/opt/fluent-bit/bin/fluent-bit -c //etc/fluent-bit/fluent-bit.yml
```

Hope it helps :)

**Comment by patrick-stephens** [comment]
@edsiper I think we should consider this for 4.0, main thing is to ensure we notify people so they can expect it.

See https://fluent-all.slack.com/archives/C0CTQGHKJ/p1736719725842959 as well.

**Comment by patrick-stephens** [comment]
I've prepped a PR for this: https://github.com/fluent/fluent-bit/pull/9837

**Comment by Elwell** [comment]
Minor typo for @P4R4DiSi4C's benefit - you can drop the double slash in the ExecStart (it's a typo upstream too) so it should be

```
[Service]
ExecStart=
ExecStart=/opt/fluent-bit/bin/fluent-bit -c /etc/fluent-bit/fluent-bit.yml

```
or .yaml depending on preferences :-)

**Comment by P4R4DiSi4C** [comment]
> Minor typo for [@P4R4DiSi4C](https://github.com/P4R4DiSi4C)'s benefit - you can drop the double slash in the ExecStart (it's a typo upstream too) so it should be
> 
> ```
> [Service]
> ExecStart=
> ExecStart=/opt/fluent-bit/bin/fluent-bit -c /etc/fluent-bit/fluent-bit.yml
> ```
> 
> or .yaml depending on preferences :-)

Thanks for that @Elwell !
